netbout-web/src/main/java/com/netbout/email/BoutInviteMail.java

```diff
@@ -0,0 +1,167 @@
+     */
+    private static final String MAIL_CONTENT =
+        "You are invited into the Netbout click on the link to register";
+
+    /**
+     * Postman.
+     */
+    private final transient Postman postman;
+
+    /**
+     * Public ctor.
+     * @param pst Postman
+     */
+    BoutInviteMail(final Postman pst) {
+        this.postman = pst;
+    }
+
+    /**
+     * Send an email with register link which is like.
+     * http://www.netbout.com/b/<bout_number>?invite=<invite-key>
+     * @param email Email
+     * @param urn Urn
+     * @param bout Bout
+     * @throws IOException if fails
+     */
+    public void send(final String email, final String urn, final Bout bout)
+        throws IOException {
+        this.postman.send(
+            new Envelope.MIME()
+                .with(new StRecipient(email))
+                .with(
+                    new StSubject(
+                        String.format(
+                            "#%d: %s",
+                            bout.number(),
+                            bout.title()
+                        )
+                    )
+                )
+                .with(
+                    new EnHTML(
+                        Joiner.on('\n').join(
+                            new Markdown(MAIL_CONTENT).html(), "<br/>",
+                            String.format(
+                                "http://www.netbout.com/b/%d?invite=%s",
+                                bout.number(),
+                                this.encode(urn)
+                            ),
+                            "<p style=\"color:#C8C8C8;font-size:2px;\">",
+                            String.format("%d</p>", System.nanoTime()),
+                            new GmailViewAction(bout.number()).xml()
+                        )
+                    )
+                )
+        );
+    }
+
+    /**
+     * Create invite key by urn using CcAES.
+     * @param urn Urn
+     * @return Encoded urn
+     * @throws IOException if fails
+     */
+    private String encode(final String urn)
+        throws IOException {
+        try {
+            final KeyGenerator generator = KeyGenerator.getInstance(ENCRYPTION);
+            generator.init(KEY_LENGTH);
+            final byte[] key = generator.generateKey().getEncoded();
+            final Codec codec = new CcAES(
+                new Codec() {
+                    @Override
+                    public Identity decode(final byte[] bytes)
+                        throws IOException {
+                        return new Identity.Simple(new String(bytes));
+                    }
+                    @Override
+                    public byte[] encode(final Identity identity)
+                        throws IOException {
+                        return identity.urn().getBytes();
+                    }
+                },
+                key
+            );
+            final byte[] encode = codec.encode(new Identity.Simple(urn));
```

Comment:
@mstzn36 will we be able to decrypt it later? where is the key stored?
I think it will be simpler to use `Jasypt` instead `CcAES`. Please see `EmCatch.encrypt()`/`decrypt()`
let's use separate secret string in `Manifest` for this.